networkx/generators/lattice.py

```diff
@@ -67,7 +67,7 @@ def grid_2d_graph(m, n, periodic=False, create_using=None):
     G.add_edges_from(((i, j), (pi, j)) for pi, i in pairwise(rows) for j in cols)
     G.add_edges_from(((i, j), (i, pj)) for i in rows for pj, j in pairwise(cols))
 
-    if iterable(periodic):
+    if hasattr(periodic, "__iter__"):
```

Comment:
Another alternative is to use the ABC's e.g. `if isinstance(periodic, Iterable)`. This would require an extra `from collections.abc import Iterable` at the top of the file though.

Reply:
Let's use something like:
```python
try:
    periodic_r, periodic_c = periodic
except TypeError:
    periodic_r = periodic_c = periodic
```